### PR TITLE
feat(website): support pagination for chat list

### PR DESCRIPTION
**Pull request by mtosity**
Issue https://github.com/LAION-AI/Open-Assistant/issues/2584
Relate to https://github.com/LAION-AI/Open-Assistant/pull/2773

Test list before ready
- [x] Should fetch more when scrolling down
- [x] Update title should function normally
- [x] Hide chat should function normally
- [x] Create a new chat should function normally

Demo

https://user-images.githubusercontent.com/40803798/233858390-019db56d-02e2-44b6-9513-9c5213019f6b.mp4

### CODE REVIEW

File: website/src/components/Chat/ChatListBase.tsx [hunk 1]
```diff
@@ -5,50 +5,65 @@ import { Plus } from "lucide-react";
...
-import useSWR from "swr";
-
 import { ChatListItem } from "./ChatListItem";
 import { CreateChatButton } from "./CreateChatButton";
 import { InferencePoweredBy } from "./InferencePoweredBy";
+import { useListChatPagination } from "./useListChatPagination";
 
 export const ChatListBase = memo(function ChatListBase({
-  chats, // TODO: can we remove this?
+  initialChats, // TODO: can we remove this?
```
Comment (mtosity) [inline website/src/components/Chat/ChatListBase.tsx:15]: This is the server side chat list response

File: website/src/pages/api/chat/index.ts [hunk 2]
```diff
@@ -13,7 +13,17 @@ const handler = withoutRole("banned", async (req, res, token) => {
     if (req.query.chat_id) {
       data = await client.get_chat(req.query.chat_id as string);
     } else {
-      data = await client.get_my_chats();
+      const params = {};
+      if (req.query.limit) {
```
Comment (notmd) [inline website/src/pages/api/chat/index.ts:17]: Can you remove this option and hard code it to this file?

**Review by notmd — COMMENTED** [review]
Thank you! Really nice work.

File: website/src/pages/api/chat/index.ts (hunk 2 above)
Comment (mtosity) [reply website/src/pages/api/chat/index.ts]: Done

File: website/src/pages/api/chat/index.ts [hunk 3]
```diff
@@ -13,7 +15,16 @@ const handler = withoutRole("banned", async (req, res, token) => {
     if (req.query.chat_id) {
       data = await client.get_chat(req.query.chat_id as string);
     } else {
-      data = await client.get_my_chats();
+      const params = {
```
Comment (AbdBarho) [inline website/src/pages/api/chat/index.ts:18]: please add type `params: GetChatParams` just to be sure.

Otherwise looks good!

Comment (mtosity) [reply website/src/pages/api/chat/index.ts]: Oh I miss type this, added

**Review by AbdBarho — APPROVED** [review]

**Review by notmd — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**